scipy/special/basic.py

```diff
@@ -2264,7 +2277,27 @@ def factorial(n, exact=False):
 
     """
     if exact:
-        return math.factorial(n)
+        if np.ndim(n) == 0:
+            return math.factorial(n)
+        else:
+            n = asarray(n)
+            un = np.unique(n)
+
+            if any(abs(un.astype(int)) != un):
```

Comment:
Could do `not np.array_equal(un.astype(int), un)` right?

Also, after this point, `un` could still be float, so when you do the multiplies above, you could end up with a problem where the floating point values multiply out, get stored in the `int` array (via truncation` and thus land at the wrong values, right? It seems like it would be clearer to actually cast `un = un.astype(int)` after this check, so that calculations afterward are guaranteed to be done with integer arithmetic.